docs/cloudinit.adoc

```diff
@@ -0,0 +1,37 @@
+= CloudInit
+:idprefix:
+:idseparator: -
+:sectlinks:
+:sectnums:
+:toc: auto
+
+:uri-cloudinit: https://docs.oracle.com/en-us/iaas/Content/ContEng/Tasks/contengusingcustomcloudinitscripts.htm
+
+== Instructions:
+
+* The default cloudinit script is as below:
+----
+#!/bin/bash
+
+# DO NOT MODIFY
+curl --fail -H "Authorization: Bearer Oracle" -L0 http://169.254.169.254/opc/v2/instance/metadata/oke_init_script | base64 --decode >/var/run/oke-init.sh
+
+## run oke provisioning script
+bash -x /var/run/oke-init.sh
+
+### adjust block volume size
+/usr/libexec/oci-growfs -y
+
+timedatectl set-timezone ${worker_timezone}
+----
+
+* To customize this you can modify the above script and pass the script as input variable to `cloudinit_nodepool_common`. This script will be used across all the nodepools.
+
+* To use specific cloud init script for a nodepool pass the script as input variable to `cloudinit_nodepool` as a map.This will take precedence over `cloudinit_nodepool_common` for that nodepool.
+Ex: cloudinit_nodepool        = {
+  #np1 = "/tmp/np1cloudinit.sh"
```

Comment:
Are your filenames just a convention or something that's expected by nodepool name?

Reply:
the filename can be anything its just an example.